iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/MandatoryLabelCheck.java

```diff
@@ -0,0 +1,120 @@
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.docker.symbols.ArgumentResolution;
+import org.sonar.iac.docker.tree.api.Body;
+import org.sonar.iac.docker.tree.api.DockerImage;
+import org.sonar.iac.docker.tree.api.LabelInstruction;
+import org.sonar.iac.docker.tree.api.OnBuildInstruction;
+
+@Rule(key = "S7028")
+public class MandatoryLabelCheck implements IacCheck {
+  private static final String MESSAGE = "Add the missing label%s: %s.";
+  private static final String DEFAULT_REQUIRED_LABELS = "maintainer";
+  private Set<String> requiredLabelSet = Set.of(DEFAULT_REQUIRED_LABELS);
+
+  @RuleProperty(
+    key = "labels",
+    description = "Comma separated list of required labels.",
+    defaultValue = DEFAULT_REQUIRED_LABELS)
+  String requiredLabels = DEFAULT_REQUIRED_LABELS;
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(Body.class, this::checkBody);
+    requiredLabelSet = splitLabels(requiredLabels);
+  }
+
+  static Set<String> splitLabels(String labels) {
+    return Arrays.stream(labels.split(","))
+      .map(String::trim)
+      .filter(s -> !s.isBlank())
+      .collect(Collectors.toSet());
+  }
+
+  private void checkBody(CheckContext ctx, Body body) {
+    Set<String> encounteredLabels = new HashSet<>();
+
+    body.dockerImages().stream()
+      .flatMap(MandatoryLabelCheck::gatherLabelInstructions)
+      .forEach(labelInstruction -> handleLabelInstruction(labelInstruction, encounteredLabels));
+
+    List<String> missingLabels = new ArrayList<>(requiredLabelSet);
+    missingLabels.removeAll(encounteredLabels);
+    if (!missingLabels.isEmpty()) {
+      if (missingLabels.size() == 1) {
+        ctx.reportIssue(body.dockerImages().get(0).from(), MESSAGE.formatted("", "\"" + missingLabels.get(0) + "\""));
+      } else {
+        ctx.reportIssue(body.dockerImages().get(0).from(), MESSAGE.formatted("s", formatLabels(missingLabels)));
+      }
+    }
+  }
+
+  private static Stream<LabelInstruction> gatherLabelInstructions(DockerImage image) {
+    return image.instructions().stream()
+      .filter(instruction -> instruction instanceof LabelInstruction
+        || (instruction instanceof OnBuildInstruction onBuildInstruction
+          && onBuildInstruction.instruction() instanceof LabelInstruction))
+      .map(instruction -> {
+        if (instruction instanceof LabelInstruction labelInstruction) {
+          return labelInstruction;
+        } else {
+          return ((LabelInstruction) ((OnBuildInstruction) instruction).instruction());
+        }
+      });
+  }
+
+  private static void handleLabelInstruction(LabelInstruction labelInstruction, Set<String> labelInstructions) {
+    labelInstruction.labels().stream()
+      .map(label -> ArgumentResolution.of(label.key()).value())
+      .forEach(labelInstructions::add);
+  }
+
+  private static String formatLabels(List<String> missingLabels) {
```

Comment:
Please add a test where there are exactly 2 missing labels (or just a test for this formatting method)